### ISSUE TITLE
Incorrect warning when using pipe in capture.

### ISSUE THREAD

**Issue by TBK145** [rrrene/credo#182]
When a piping sequence starts with a capture like &1, mix credo wrongfully reports that a piping chain should start with a raw value.
```elixir
defmodule Test do
  def test do
    foo = %{bar: %{}}
    update_in foo.bar, &(
      &1
      |> Map.put(:a, :b))
  end
end
```
```
  Refactoring opportunities                                                     
┃ 
┃ [F] → Pipe chain should start with a raw value.
┃       test.ex:6 (Test.test)
```

**Comment by rrrene** [comment]
Thx for reporting this. I added a test reproducing the error :+1:

**Comment by sstern6** [comment]
@rrrene See your tests, about to make 2 PR's, one for `&`, the other for `++` just to keep things separate.  If I need to go deeper for any reason we can hash that out on the PR.  However, it simply seems like an atom representation of `&` and `++` need to be included here:

https://github.com/rrrene/credo/blob/master/lib/credo/check/refactor/pipe_chain_start.ex#L34

Thanks for including the tests!
